Sources/Datadog/RUM/RUMEvent/RUMEventBuilder.swift

```diff
@@ -7,26 +7,23 @@
 import Foundation
 
 internal class RUMEventBuilder {
-    let userInfoProvider: UserInfoProvider
     let eventsMapper: RUMEventsMapper
 
-    init(
-        userInfoProvider: UserInfoProvider,
-        eventsMapper: RUMEventsMapper
-    ) {
-        self.userInfoProvider = userInfoProvider
+    init(eventsMapper: RUMEventsMapper) {
         self.eventsMapper = eventsMapper
     }
 
     func createRUMEvent<DM: RUMDataModel>(
         with model: DM,
         attributes: [String: Encodable]
     ) -> RUMEvent<DM>? {
-        let event = RUMEvent(
-            model: model,
-            attributes: attributes,
-            userInfoAttributes: userInfoProvider.value.extraInfo
-        )
+        var model = model
+
+        if !attributes.isEmpty {
+            model.context = RUMEventAttributes(contextInfo: attributes)
```

Comment:
There's one last thing to cleanup. Similar to how we emit `RUMUser` in RUM scopes, we can now emit `RUMEventAttributes` right in scopes:
```swift
// RUMViewScope.swift
// RUMResourceScope.swift
// RUMUserActionScope.swift

let eventData = RUM####Event(
   // ...
   usr: dependencies.userInfoProvider.current
   // ...
   context: .init(contextInfo: attributes),
   // ...
)

let event = dependencies.eventBuilder.createRUMEvent(with: eventData)
```
That will polish the architecture by (finally 🚀) building RUM events only in RUM scopes ✨.

This change should be relatively easy and safe as all RUM scope tests use the actual models and we've already updated assertions in this PR.

Reply:
Awesome, will do! 🧹